packages/common/src/pipes/async_pipe.ts

```diff
@@ -94,10 +94,7 @@ export class AsyncPipe implements OnDestroy, PipeTransform {
     }
   }
 
-  transform<T>(obj: null): null;
-  transform<T>(obj: undefined): undefined;
-  transform<T>(obj: Observable<T>|null|undefined): T|null;
-  transform<T>(obj: Promise<T>|null|undefined): T|null;
+  transform<T>(obj: Observable<T>|Promise<T>|null|undefined): T|null;
   transform(obj: Observable<any>|Promise<any>|null|undefined): any {
```

Comment:
since the object wrapping has been removed, we might avoid even just use the "strict" signature, if we like it better:

```suggestion
  transform<T>(obj: Observable<T>|Promise<T>|null|undefined): T|null {
```

note that this would not change anything for the API consumers